kolibri/plugins/coach/assets/src/views/common/assignments/SidePanelRecipientsSelector/LearnersSelectorSidePanel.vue

```diff
@@ -0,0 +1,192 @@
+      },
+      adHocLearners: {
+        type: Array,
+        required: false,
+        default: () => [],
+      },
+      selectedGroupIds: {
+        type: Array,
+        required: true,
+      },
+      disabled: {
+        type: Boolean,
+        default: false,
+      },
+      classId: {
+        type: String,
+        required: true,
+      },
+    },
+    data() {
+      return {
+        workingAdHocLearners: this.adHocLearners,
+        workingSelectedGroupIds: this.selectedGroupIds,
+      };
+    },
+    computed: {
+      ...mapGetters('classSummary', ['learners']),
+      ...mapState('classSummary', ['groupMap']),
+      ungroupedLearnersIds() {
+        return this.learners
+          .filter(learner => {
+            return Object.values(this.groupMap).every(
+              group => !group.member_ids.includes(learner.id),
+            );
+          })
+          .map(learner => learner.id);
+      },
+      allUngroupedLearnresIsSelected() {
+        return this.ungroupedLearnersIds.every(learnerId =>
+          this.workingAdHocLearners.includes(learnerId),
+        );
+      },
+    },
+    methods: {
+      focusFirstEl() {
+        this.$nextTick(() => {
+          this.$el.querySelector('input').focus();
+        });
+      },
+      groupIsSelected({ id }) {
+        return this.workingSelectedGroupIds.includes(id);
+      },
+      toggleGroup(isChecked, { id }) {
+        if (isChecked) {
+          this.workingSelectedGroupIds = [...this.workingSelectedGroupIds, id];
+        } else {
+          this.workingSelectedGroupIds = this.workingSelectedGroupIds.filter(
+            groupId => groupId !== id,
+          );
+        }
+      },
+      updateAdHocLearners(learnerIds) {
+        this.workingAdHocLearners = learnerIds;
+      },
+      selectAllUngroupedLearners(isChecked) {
+        if (isChecked) {
+          this.workingAdHocLearners = uniq([
+            ...this.workingAdHocLearners,
+            ...this.ungroupedLearnersIds,
+          ]);
+        } else {
+          this.workingAdHocLearners = this.workingAdHocLearners.filter(
+            learnerId => !this.ungroupedLearnersIds.includes(learnerId),
+          );
+        }
+      },
+      save() {
+        this.$emit('update:adHocLearners', this.workingAdHocLearners);
+        this.$emit('update:selectedGroupIds', this.workingSelectedGroupIds);
+        this.$emit('close');
+      },
+    },
+    $trs: {
+      allUngroupedLearnres: {
+        message: 'All Ungrouped Learners',
+        context: 'Option to select all learners that are not in a group',
+      },
+      selectGroupsAndIndividualLearnersTitle: {
+        message: 'Select groups and individual learners',
+        context: 'Title for the side panel to select groups and individual learners',
+      },
+    },
+  };
+
+</script>
+
+
+<style lang="scss" scoped>
+
+  .bottom-nav-container {
+    display: flex;
+    justify-content: flex-end;
+    width: 100%;
+    margin-bottom: 16px;
```

Comment:
Once #12895 is merged we will need to remove this maring, as its adding a fixing-marging that wont be needed after #12895